python/ray/util/client/worker.py

```diff
@@ -190,51 +189,26 @@ def connection_info(self):
     def register_callback(
             self, ref: ClientObjectRef,
             callback: Callable[[ray_client_pb2.DataResponse], None]) -> None:
-        req = ray_client_pb2.GetRequest(id=ref.id, asynchronous=True)
+        req = ray_client_pb2.GetRequest(ids=[ref.id], asynchronous=True)
         self.data_client.RegisterGetCallback(req, callback)
 
     def get(self, vals, *, timeout: Optional[float] = None) -> Any:
-        to_get = []
-        single = False
         if isinstance(vals, list):
             to_get = vals
+            single = False
         elif isinstance(vals, ClientObjectRef):
             to_get = [vals]
             single = True
         else:
             raise Exception("Can't get something that's not a "
                             "list of IDs or just an ID: %s" % type(vals))
-        if timeout is None:
-            timeout = 0
-            deadline = None
-        else:
-            deadline = time.monotonic() + timeout
-        out = []
-        for obj_ref in to_get:
-            res = None
-            # Implement non-blocking get with a short-polling loop. This allows
-            # cancellation of gets via Ctrl-C, since we never block for long.
-            while True:
-                try:
-                    if deadline:
-                        op_timeout = min(
```

Comment:
Can we keep the `MAX_BLOCKING_OPERATION`. This is still relevant because the server will block for the duration of the Ray Get!

Reply:
Misunderstood how threading.Condition() works. Reverted this part. Will convert this to asyncio in a subsequent pr.